qualtran/bloqs/arithmetic/controlled_addition.py

```diff
@@ -91,50 +101,23 @@ def signature(self):
             [Register("ctrl", QBit()), Register("a", self.a_dtype), Register("b", self.b_dtype)]
         )
 
-    def add_my_tensors(
-        self,
-        tn: 'qtn.TensorNetwork',
-        tag: Any,
-        *,
-        incoming: Dict[str, 'SoquetT'],
-        outgoing: Dict[str, 'SoquetT'],
-    ):
-        import quimb.tensor as qtn
-
-        if isinstance(self.a_dtype, QInt) or isinstance(self.b_dtype, QInt):
-            raise TypeError("Tensor contraction for addition is only supported for unsigned ints.")
-        N_a = 2**self.a_dtype.bitsize
-        N_b = 2**self.b_dtype.bitsize
-        inds = (
-            incoming['ctrl'],
-            incoming['a'],
-            incoming['b'],
-            outgoing['ctrl'],
-            outgoing['a'],
-            outgoing['b'],
-        )
-        unitary = np.zeros((2, N_a, N_b, 2, N_a, N_b), dtype=np.complex128)
-        for c, a, b in itertools.product(range(2), range(N_a), range(N_b)):
-            if c == self.cv:
-                unitary[c, a, b, c, a, int(math.fmod(a + b, N_b))] = 1
-            else:
-                unitary[c, a, b, c, a, b] = 1
-
-        tn.add(qtn.Tensor(data=unitary, inds=inds, tags=[self.short_name(), tag]))
-
-    def decompose_bloq(self) -> 'CompositeBloq':
-        return decompose_from_cirq_style_method(self)
-
     def on_classical_vals(self, **kwargs) -> Dict[str, 'ClassicalValT']:
         a, b = kwargs['a'], kwargs['b']
-        unsigned = isinstance(self.a_dtype, (QUInt, QMontgomeryUInt))
-        b_bitsize = self.b_dtype.bitsize
-        N = 2**b_bitsize if unsigned else 2 ** (b_bitsize - 1)
         ctrl = kwargs['ctrl']
         if ctrl != self.cv:
             return {'ctrl': ctrl, 'a': a, 'b': b}
         else:
-            return {'ctrl': ctrl, 'a': a, 'b': int(math.fmod(a + b, N))}
+            if not isinstance(self.b_dtype.bitsize, int):
+                raise NotImplementedError(
```

Comment:
we should only use `NotImplementedError` for cases that are theoretically possible but we just haven't done it _yet_. Is this the case here? If it's an error in any future, I'd suggest `ValueError`.

Reply:
changed to ValueError ... from a code point of view a sympy symbol bitsize also works for this specific operation but it fails other classical operations ... this is why I wasn't sure